app/workers/fast_track/disability_compensation_job.rb

```diff
@@ -79,16 +81,19 @@ def send_fast_track_engineer_email_for_testing(form526_submission_id, error_mess
     end
 
     def get_icn(form526_submission)
-      account(form526_submission).icn.presence
+      account_record = account(form526_submission)
+      raise AccountNotFoundError, "for user_uuid: #{form526_submission.user_uuid} or their edipi" unless account_record
+
+      account_record.icn.presence
     end
 
     def account(form526_submission)
       user_uuid = form526_submission.user_uuid.presence
       edipi = form526_submission.auth_headers['va_eauth_dodedipnid'].presence
       # rubocop:disable Lint/UselessAssignment
-      account = Account.where(idme_uuid: user_uuid).first if user_uuid
-      account ||= Account.where(logingov_uuid: user_uuid).first if user_uuid
-      account ||= Account.where(edipi: edipi).first if edipi
+      account = Account.find_by(idme_uuid: user_uuid) if user_uuid
+      account ||= Account.find_by(logingov_uuid: user_uuid) if user_uuid
+      account ||= Account.find_by(edipi: edipi) if edipi
```

Comment:
I got inconsistent failures with this change but it eventually passed. `find_by` should call `where`, which is set up to return `Account.none` in [the test](https://github.com/department-of-veterans-affairs/vets-api/pull/9024/files#diff-45106a8a44ee0d5f717fd239680d2e9bd7853c549f03ec4ae26f094d207c63e2R159).

Reply:
I ended up adding `allow(Account).to receive(:find_by).and_return nil` to the test setup due to [this test failure](https://github.com/department-of-veterans-affairs/vets-api/pull/9024/checks?check_run_id=5058823500)

Reply:
One difference is `where` orders the results but this is irrelevant for `idme_uuid` and `logingov_uuid` because they have DB uniqueness constraints. But there is an argument to use `where` for `edipi` but is ordering by `id` appropriate?

```ruby
Account.find_by(idme_uuid: "asdf")
2022-02-03 15:57:34.147248 D [18204:67400 log_subscriber.rb:126] ActiveRecord::Base --   Account Load (13.6ms)  SELECT "accounts".* FROM "accounts" WHERE "accounts"."idme_uuid" = $1 LIMIT $2  [["idme_uuid", "asdf"], ["LIMIT", 1]]
=> nil

Account.where(idme_uuid: "asdf").first
2022-02-03 15:57:47.214873 D [18204:67400 log_subscriber.rb:126] ActiveRecord::Base --   Account Load (7.1ms)  SELECT "accounts".* FROM "accounts" WHERE "accounts"."idme_uuid" = $1 ORDER BY "accounts"."id" ASC LIMIT $2  [["idme_uuid", "asdf"], ["LIMIT", 1]]
```